src/source/source_simple_transistor.ts

```diff
@@ -0,0 +1,36 @@
+import { z } from "zod"
+import {
+  source_component_base,
+  type SourceComponentBase,
+} from "src/source/base/source_component_base"
+import { voltage, current } from "src/units"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_transistor = source_component_base.extend({
+  ftype: z.literal("simple_transistor"),
+  transistorType: z.enum(["NPN", "PNP"]),
+  vce: voltage.optional(),
+  vbe: voltage.optional(),
+  ic: current.optional(),
```

Comment:
don't abbreviate, also i don't know what these are so once you unabbreviate there's a chance they're not needed (can't know unless they are fixed)